samples/iolib/src/main/cpp/player/SimpleMultiPlayer.cpp

```diff
@@ -48,9 +50,10 @@ DataCallbackResult SimpleMultiPlayer::onAudioReady(AudioStream *oboeStream, void
 
     memset(audioData, 0, numFrames * sizeof(float));
 
+    OneShotSampleSource *** sources = mSampleSources.get();
```

Comment:
Triple *!

Reply:
That seems to be how that unique_ptr<> works. Maybe Mikhail can provide some insight?

Reply:
Paul, you are using an incorrect type for your unique_ptrs. If I understand it correctly, you have an array of pointers. For that, your unique_ptr must be declared as: std::unique_ptr<SampleBuffer*[]>. Please do not decay an array into a pointer because then unique_ptr will use 'delete' instead of 'delete[]' for an array, and that's a memory leak.

BTW, who manages these SampleBuffer instances that we are pointing to? Why it's OK to use raw pointers?

Reply:
Would it be better to have a unique_ptr to an array of unique_ptrs to SampleBuffers?

Reply:
I will revisit this.